src/Time/Units.hs

```diff
@@ -163,6 +170,16 @@ instance NFData (Time (rat :: Rat))
 instance Serialise (Time (rat :: Rat))
 #endif
 
+#ifdef HAS_aeson
+instance (KnownUnitName unit) => ToJSON (Time (unit :: Rat)) where
+    toJSON = toJSON . show
+
+instance (KnownUnitName unit) => FromJSON (Time (unit :: Rat)) where
+    parseJSON = withText "time" $ \t -> fromMaybe
+        (fail $ "Can not parse Time. Expected unit: " ++ unitNameVal @unit)
+            $ pure <$> (readMaybe @(Time unit) $ Text.unpack t)
```

Comment:
I think this can be written a little bit shorter and probably clearer with the help of `maybe` function. 

`fromMaybe x $ pure <$> t` is equivalent to `maybe x pure t`. Unfortunately, `base` package doesn't have `whenNothing` combinator which is exactly `maybe x pure t` :disappointed: 

You also can go with pointfree and write something like `maybe . readMaybe . Text.unpack` (but maybe this will be less clearer.